From 71c92e8a88e6d8a59f02ab3c9691afcec979ac75 Mon Sep 17 00:00:00 2001
From: Aaron He <aaron@bluestacks.com>
Date: Thu, 17 Aug 2023 18:00:44 +0800
Subject: [PATCH 005/224] [A13] Temporarily disable KeyStore function to avoid
 system crash

Error log:

E AndroidRuntime: *** FATAL EXCEPTION IN SYSTEM PROCESS: LockSettingsService
E AndroidRuntime: java.lang.AssertionError: 4
E AndroidRuntime:        at android.security.KeyStore.state(KeyStore.java:67)
E AndroidRuntime:        at com.android.server.locksettings.LockSettingsService.ensureProfileKeystoreUnlocked(LockSettingsService.java:789)
E AndroidRuntime:        at com.android.server.locksettings.LockSettingsService.-$$Nest$mensureProfileKeystoreUnlocked(Unknown Source:0)
E AndroidRuntime:        at com.android.server.locksettings.LockSettingsService$1.run(LockSettingsService.java:807)
E AndroidRuntime:        at android.os.Handler.handleCallback(Handler.java:942)
E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:99)
E AndroidRuntime:        at android.os.Looper.loopOnce(Looper.java:201)
E AndroidRuntime:        at android.os.Looper.loop(Looper.java:288)
E AndroidRuntime:        at android.os.HandlerThread.run(HandlerThread.java:67)
E AndroidRuntime:        at com.android.server.ServiceThread.run(ServiceThread.java:44)
---
 .../server/locksettings/LockSettingsService.java   | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/services/core/java/com/android/server/locksettings/LockSettingsService.java b/services/core/java/com/android/server/locksettings/LockSettingsService.java
index 78cffa6f4f79..89acd33f5add 100644
--- a/services/core/java/com/android/server/locksettings/LockSettingsService.java
+++ b/services/core/java/com/android/server/locksettings/LockSettingsService.java
@@ -397,7 +397,8 @@ public class LockSettingsService extends ILockSettings.Stub {
         // This can only happen during an upgrade path where SID is yet to be
         // generated when the user unlocks for the first time.
         try {
-            if (getGateKeeperService().getSecureUserId(parentId) == 0) {
+
+            if (getGateKeeperService() != null && getGateKeeperService().getSecureUserId(parentId) == 0) {
                 return;
             }
         } catch (RemoteException e) {
@@ -785,6 +786,7 @@ public class LockSettingsService extends ILockSettings.Stub {
      * during the normal flow. In this case unlock the keystore for the profile.
      */
     private void ensureProfileKeystoreUnlocked(int userId) {
+    /*
         final KeyStore ks = KeyStore.getInstance();
         if (ks.state(userId) == KeyStore.State.LOCKED
                 && isCredentialSharableWithParent(userId)
@@ -793,8 +795,10 @@ public class LockSettingsService extends ILockSettings.Stub {
             // If boot took too long and the password in vold got expired, parent keystore will
             // be still locked, we ignore this case since the user will be prompted to unlock
             // the device after boot.
-            unlockChildProfile(userId, true /* ignoreUserNotAuthenticated */);
-        }
+            unlockChildProfile(userId, true);
+        } */
+           // unlockChildProfile(userId, true /* ignoreUserNotAuthenticated */);
+
     }
 
     public void onUnlockUser(final int userId) {
@@ -3571,11 +3575,13 @@ public class LockSettingsService extends ILockSettings.Stub {
             if (mDeviceProvisionedUri.equals(uri)) {
                 updateRegistration();
 
+                /*
                 if (isProvisioned()) {
                     Slog.i(TAG, "Reporting device setup complete to IGateKeeperService");
                     reportDeviceSetupComplete();
                     clearFrpCredentialIfOwnerNotSecure();
                 }
+                */
             }
         }
 
@@ -3595,11 +3601,13 @@ public class LockSettingsService extends ILockSettings.Stub {
         }
 
         private void reportDeviceSetupComplete() {
+            /*
             try {
                 getGateKeeperService().reportDeviceSetupComplete();
             } catch (RemoteException e) {
                 Slog.e(TAG, "Failure reporting to IGateKeeperService", e);
             }
+            */
         }
 
         /**
